src/bms_sender.py

```diff
@@ -10,24 +11,48 @@ def __init__(self):
                                     stdout=subprocess.PIPE).communicate()[0].rstrip().decode('utf-8')
         self.sender_tc_json_path = os.path.join(self.root_dir, "inc", "sender_test_case.json")
         self.sender_temp_limit_json_path = os.path.join(self.root_dir, "inc", "sender_temperature_limit.json")
-        self.sender_console_out_json_path = os.path.join(self.root_dir, "inc", "sender_console_output.json") 
+        self.test_regex = r"Sender\s\d+\s\-\sB\d+\s[\w\-\s\,]+\|"
         
     def get_json_data(self, path):
         data = {}
         if os.path.exists(path):
             with open(path, 'r') as jsonread:
                 data = json.load(jsonread)
         return data
-
-    def celcius_to_Fahr(self, C_temp):
-        # Convert Celcius to Fahrenheit
-        F_temp = (C_temp*(9/5)) + 32
-        return F_temp
-    
-    def Fahr_to_celcius(self, F_temp):
-        # Convert Fahrenheit to Celcius
-        C_temp = (F_temp - 32)*(5/9)
-        return C_temp
     
     def create_random_value(self, min, max):
         return random.randint(min, max)
+    
+    def select_random_battery(self, no_of_sender_battery):
+        return self.create_random_value(0, no_of_sender_battery-1)
+        
+    def create_temp_stream_output(self, idx, battery_idx, bat):
+        temp_limit_json_data = self.get_json_data(self.sender_temp_limit_json_path)
+        min_temp = temp_limit_json_data["Temperature_Limit"]["Min"]
+        max_temp = temp_limit_json_data["Temperature_Limit"]["Max"]
+        rand_temp_out = self.create_random_value(min_temp, max_temp)
+        out_data = "Sender {} - B{} - {}, {}, {} |".format(battery_idx+1, idx+1, bat["Battery_Type"], bat["Temp_Type"], rand_temp_out)
+        print(out_data)
+        return out_data
+    
+    def create_stream_output(self, test_case_json_data):
```

Comment:
code is not simulating 50 values stream as per below requirement.
	"simulates and sends at least two Battery / Charging parameters
	sends fifty readings in a stream"

Reply:
Hi @sunilgowda95, the streaming output is visible in the Action -> https://github.com/clean-code-craft-tcq-7/stream-line-omprakashs855/actions/runs/4194555799/jobs/7272875583
do check the attached image for more info,
as due to multiple cases, multiple series of output is visible
the streaming count can be modified using https://github.com/clean-code-craft-tcq-7/stream-line-omprakashs855/blob/master/inc/sender_test_case.json
file

Reply:
The output can be seen for two batteries selected randomly showing two battery parameters data (temp, soc)
https://github.com/clean-code-craft-tcq-7/stream-line-omprakashs855/actions/runs/4222298284/jobs/7330633400
Please check

Reply:
Hi,
Yes I can see data from 2 batteries taken randomly and sending output stream totaling to 50.

Shouldn't sender send 50values of battery parameters from same battery?
That constant stream of data for each sensor from sender will help in doing better analysis at receiver.

Reply:
Hi @sunilgowda95 , the streaming data can be from any batteries (given at least 2), currently we should focus on piping the data and creating two statistics ( one for sma, and one for min/max)
Further changes can be done if needed
FYI @sudeeprp

Reply:
Hi @omprakashs855 
Following are the requirements given,

The Sender

    simulates and sends at least two Battery / Charging parameters
    sends fifty readings in a stream
    can either generate values for the parameters, or read from a file
    uses console output to communicate the parameters.

The Receiver

    reads the parameters from the console input
    after reading every parameter, it prints the following:
        maximum and minimum values in the incoming stream
        [simple moving average](https://www.investopedia.com/terms/s/sma.asp) of the last 5 values

As you can clearly see sender is required to send 50 reading of at-least 2 parameters,
and receiver has to use at-least 5 values to calculate analytics.

With randomly selecting sensors we may also get a situation where 5 values are not available to calculate analytics for a sensor.

Reply:
Hi @sunilgowda95 , as you can see from the receiver requirement, we have to pipe the output for the incoming Stream
so at its first value, min = max = avg
then the second value, and then the rest ...

we need to take stream of sender output, and incoming stream input for receiver

Reply:
Hi @omprakashs855 I have tested receiver with most possible scenarios using test cases. Here I was trying to infer on randomly selecting sensors and simulating data. In some corner cases with less than necessary amount of data received by receiver, the analytics may go wrong.